modules/distributor/receiver/middleware.go

```diff
@@ -61,12 +61,14 @@ func (m *multiTenancyMiddleware) Wrap(next consumer.Traces) consumer.Traces {
 			info := client.FromContext(ctx)
 			orgIDs := info.Metadata.Get(user.OrgIDHeaderName)
 			if len(orgIDs) == 0 {
-				log.Logger.Log("msg", "failed to extract org id from both grpc and HTTP", "err", err)
+				log.Logger.Log("msg", "failed to extract org id from both grpc and HTTP",
+					"err", err, "client", info.Addr.String())
```

Comment:
This is causing a panic when `info.Addr == nil`. You need to add a check for that.

Reply:
HI @mapno, 
Thanks for the reviews, for version [v0.74.0](https://github.com/grafana/tempo/blob/main/go.mod#LL73C13-L73C13) of the opentelemetry-collector SDK, `info.Addr` is not a pointer. we should pay attention to future updates.

https://github.com/open-telemetry/opentelemetry-collector/blob/v0.74.0/client/client.go#L104

Reply:
`net.Addr` is an interface, so it panics when undefined. See example https://go.dev/play/p/dx6a1BYB16J.

Reply:
@mapno Thanks for the correction, the panics has been fixed